### ISSUE TITLE
Performance degradation while producing messages to Kafka

### ISSUE THREAD

**Issue by pcasaes** [smallrye/smallrye-reactive-messaging#631]
While testing Quarkus 1.6.0.CR1 I noticed a large performance hit while producing Kafka messages in comparison with 1.5.2.Final (smallrye reactive messaging 2.0.3 vs 2.1.0). The impact expressed itself as an order of magnitude increase in latency and CPU. 

I was able to track it down to this commit:
e439631487a76fdae84b7ae88cb0c3c59f613ef9

Lines 80-89 on KafkaSink.java
```
                return Uni.createFrom()
                        .<Void> emitter(
                                e -> stream.send((ProducerRecord) record, ar -> handleWriteResult(ar, message, record, e)))
                        .onFailure().retry()
                        .withBackOff(Duration.ofSeconds(1), Duration.ofSeconds(20)).atMost(config.getRetries())
                        .onFailure().recoverWithUni(t -> {
                            // Even with the retry we still fail, nack the message.
                            log.nackingMessage(message, actualTopic, t);
                            return Uni.createFrom().completionStage(message.nack(t));
                        });
```

I reverted just these lines locally and the performance penalty disappeared 
```
                return Uni.createFrom()
                        .emitter(e -> stream.write((ProducerRecord) record, ar -> handleWriteResult(ar, message, record, e)));
```

It is generally not recommended to retry producing Kafka messages. Doing so can lead to duplicate messages since the Kafka java library already retries indefinitely (by default).

I have Flamegraphs which I'll make available on the mailing group.

**Comment by cescoffier** [comment]
These lines were added to handle the case where you don't have a Kafka broker ready (and so no topic in topology yet). It might be only necessary on startup.

Another solution is to introduce the long time overdue readiness check which will hold sending until we are ready.

**Comment by cescoffier** [comment]
Fixed by #632
